### ISSUE TITLE
Error: File must be file instance with ZIP extension

### ISSUE THREAD

**Issue by Harsturomai** [opencv/cvat#5274]
### My actions before raising this issue
- [x] Read/searched [the docs](https://github.com/cvat-ai/cvat/tree/master#documentation)
- [x] Searched [past issues](/issues)

### Expected Behaviour
Expected to be able to upload custom annotations that aren't necessarily  `.zip`, especially when the signature for adding an annotation import allows for different extensions.

`@importer(name="MyFormat", version="2.0", ext="XML")`

### Current Behaviour
Throws `Error: File must be file instance with ZIP extension` when trying to upload.

### Possible Solution
It seems to be a validation bug on the UI side. Can be fixed by removing validation step `cvat-core/src/annotations.ts:line 312`

`throw new ArgumentError('File must be file instance with ZIP extension');`

I may well be missing something in the docs here, but even if I am, I'd like to just be able to upload a single json file that contains annotations in a custom format.

**Comment by zhiltsov-max** [comment]
@bsekachev , please look at this issue. It seems that UI imposes extra limitations, the server supports non-zip files in several formats.

**Comment by Harsturomai** [comment]
Thanks for looking into this. Please note also, that when exporting datasets using `@exporter` without saving images, the UI also asserts saving a `.zip` file regardless of the extension specified in the function. I won't create a separate issue as these are tightly coupled. I similarly hot-patched this - but would be good to see a proper solution.

Can I ask, how will I know when such fixes are implemented? Will they be pushed to the develop branch or will a separate branch be created?

**Comment by zhiltsov-max** [comment]
> Please note also, that when exporting datasets using @exporter without saving images, the UI also asserts saving a .zip file regardless of the extension specified in the function.

I wouldn't consider this a bug, because currently we always expect archives from the export. They may contain just a single file in some cases (e.g. TF export of a project with 1 subset, CVAT for * with 1 project subset and only annotations), but in this case the limitation is intended.

> Can I ask, how will I know when such fixes are implemented?

Typically, we link PRs to issues they affect, so the issues get closed after merging or, at least, be referenced in the discussion.

> Will they be pushed to the develop branch or will a separate branch be created?

We prepare fixes in separate branches, then merge into develop.

**Comment by Harsturomai** [comment]
Makes sense, thanks. One comment on the export. When CVAT saves the export as a `.zip`, the single JSON file I return from my exporter (with `EXT=JSON`) is saved as json, but with a `.zip` extension. Therefore, I cannot extract the file, but have to replace the `.zip` with `.json`. Perhaps either the exporter should be required to return a `.zip`, or CVAT should internally wrap it up into one.
